### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by Vitomir2** [graphprotocol/graph-node#5708]
### What is the current behavior?

I am using the docker configuration and the Graph node container crashes with error `exited with code 0` while the other two - postgres and ipfs are running fine. I am using apple silicon and followed the [instructions here](https://github.com/graphprotocol/graph-node/tree/master/docker#running-graph-node-on-an-macbook-m1).

I have checked if the postgres and ipfs containers are running fine and they are both good. Additionally, checked if the graph-node user exists in the postgres and it is also there. I have tested to login with the `let-me-in` pass and it works fine, too.

### Docker general settings:

Using the newer version of docker compose: `docker compose up`, without hyphen (-)

#### Docker resource settings:

CPUs: 8
Memory: 8GB
Swap: 1GB

#### What is the expected behavior?
I am expecting the graph-node container to run properly, too.

### Relevant log output

```Shell
docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
[+] Running 3/3
 ✔ Container docker-ipfs-1        Created                                                                         0.0s 
 ✔ Container docker-graph-node-1  Created                                                                         0.0s 
 ✔ Container docker-postgres-1    Recreated                                                                       0.1s 
Attaching to graph-node-1, ipfs-1, postgres-1
ipfs-1        | Changing user to ipfs
ipfs-1        | ipfs version 0.17.0
ipfs-1        | Found IPFS fs-repo at /data/ipfs
ipfs-1        | Initializing daemon...
ipfs-1        | Kubo version: 0.17.0-4485d6b
ipfs-1        | Repo version: 12
ipfs-1        | System version: arm64/linux
ipfs-1        | Golang version: go1.19.1
graph-node-1  | PATH=/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
graph-node-1  | HOSTNAME=4309e5965c98
graph-node-1  | GRAPH_LOG=debug
graph-node-1  | postgres_host=postgres
graph-node-1  | postgres_user=graph-node
graph-node-1  | postgres_pass=let-me-in
graph-node-1  | postgres_db=graph-node
graph-node-1  | ipfs=ipfs:5001
graph-node-1  | ethereum: 'mainnet:http://host.docker.internal:8545'
graph-node-1  | RUSTUP_HOME=/usr/local/rustup
graph-node-1  | CARGO_HOME=/usr/local/cargo
graph-node-1  | RUST_VERSION=1.82.0
graph-node-1  | HOME=/root
ipfs-1        | 2024/11/19 13:37:49 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
graph-node-1 exited with code 0
postgres-1    | 
postgres-1    | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres-1    | 
postgres-1    | 2024-11-19 13:37:49.715 UTC [1] LOG:  starting PostgreSQL 17.1 (Debian 17.1-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres-1    | 2024-11-19 13:37:49.715 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres-1    | 2024-11-19 13:37:49.715 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-1    | 2024-11-19 13:37:49.716 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-1    | 2024-11-19 13:37:49.722 UTC [29] LOG:  database system was shut down at 2024-11-19 11:52:18 UTC
postgres-1    | 2024-11-19 13:37:49.731 UTC [1] LOG:  database system is ready to accept connections
ipfs-1        | Swarm listening on /ip4/127.0.0.1/tcp/4001
ipfs-1        | Swarm listening on /ip4/127.0.0.1/udp/4001/quic
ipfs-1        | Swarm listening on /ip4/172.18.0.2/tcp/4001
ipfs-1        | Swarm listening on /ip4/172.18.0.2/udp/4001/quic
ipfs-1        | Swarm listening on /ip6/::1/tcp/4001
ipfs-1        | Swarm listening on /ip6/::1/udp/4001/quic
ipfs-1        | Swarm listening on /p2p-circuit
ipfs-1        | Swarm announcing /ip4/127.0.0.1/tcp/4001
ipfs-1        | Swarm announcing /ip4/127.0.0.1/udp/4001/quic
ipfs-1        | Swarm announcing /ip4/172.18.0.2/tcp/4001
ipfs-1        | Swarm announcing /ip4/172.18.0.2/udp/4001/quic
ipfs-1        | Swarm announcing /ip6/::1/tcp/4001
ipfs-1        | Swarm announcing /ip6/::1/udp/4001/quic
ipfs-1        | API server listening on /ip4/0.0.0.0/tcp/5001
ipfs-1        | WebUI: http://0.0.0.0:5001/webui
ipfs-1        | Gateway (readonly) server listening on /ip4/0.0.0.0/tcp/8080
ipfs-1        | Daemon is ready
```

### Some information to help us out

- [ ] Tick this box if this bug is caused by a regression found in the latest release.
- [ ] Tick this box if this bug is specific to the hosted service.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### OS information

macOS

**Comment by andrewmatte** [comment]
Hi, I am new to the project. It looks like it's running as a daemon to me.
exited with code 0 means no error, executed as expected as a convention.

Has anything changed since you posted this? Did you try to check the server that says it's listening on /ip4/0.0.0.0/tcp/8080 ?

**Comment by andrewmatte** [comment]
Sorry, I found your bug:
 2024/11/19 13:37:49 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
graph-node-1 exited with code 0

**Comment by andrewmatte** [comment]
Here is where that page got me:

"non-BSD

It is recommended to increase the maximum buffer size by running:

sysctl -w net.core.rmem_max=7500000"

**Comment by Vitomir2** [comment]
Hey, @andrewmatte,

I apologize for my silence, but I was quite busy and actually got success after 8 hours of struggling with the problem before that. I decided to directly try to run the docker configuration on a cloud server and there was no problem. I am not sure where I screwed it up. However, now I am able to run it even on my Mac M1 and I am not required to build anything additional. Just cloned the fresh repo and then ran the `docker compose up` inside the `docker` directory.
